### PR TITLE
Allocate AgentSchedulingCustomization only when data is present

### PR DESCRIPTION
**Pull request by thardeck**
The Fleet cluster reconciler always allocated `AgentSchedulingCustomization`
as a non-nil empty pointer, even when no scheduling overrides were configured.
This caused `agentSchedulingCustomization: {}` to be written to etcd, and
OpenAPI-aware clients (Rancher UI) then rendered the field uncommented as if
it were required.

- Declare `schedulingCustomization` as a nil pointer and populate it only
  when at least one sub-field (`PodDisruptionBudget` or `PriorityClass`) is
  non-nil. A temporary variable is used to avoid assigning an empty struct when
  `SchedulingCustomization` is set but carries no data.
- Add test cases for nil, empty, single-sub-field (PDB only, PriorityClass
  only), and fully populated scheduling customization.

## Issue: 
#53781

### CODE REVIEW

File: pkg/controllers/provisioningv2/fleetcluster/fleetcluster.go [hunk 1]
```diff
@@ -250,18 +250,22 @@ func (h *handler) createCluster(cluster *provv1.Cluster, status provv1.ClusterSt
...
-			schedulingCustomization.PriorityClass = &fleet.PriorityClassSpec{}
-			schedulingCustomization.PriorityClass.PreemptionPolicy = sc.PriorityClass.PreemptionPolicy
-			schedulingCustomization.PriorityClass.Value = sc.PriorityClass.Value
+			tmp.PriorityClass = &fleet.PriorityClassSpec{}
+			tmp.PriorityClass.PreemptionPolicy = sc.PriorityClass.PreemptionPolicy
+			tmp.PriorityClass.Value = sc.PriorityClass.Value
+		}
+		if tmp.PodDisruptionBudget != nil || tmp.PriorityClass != nil {
+			schedulingCustomization = tmp
 		}
```
Comment (weyfonk) [inline pkg/controllers/provisioningv2/fleetcluster/fleetcluster.go]: nit: this could be done without `tmp`, by directly initialising `schedulingCustomization` to an empty `fleet.AgentSchedulingCustomization{}`, and populating it, only if the pod disruption budget or priority class is non-nil.
